psl-core/src/main/java/org/linqs/psl/util/MathUtils.java

```diff
@@ -155,4 +155,50 @@ public static BigInteger factorial(BigInteger number) {
 
         return result;
     }
+
+    /**
+     * Scale n-dimensional double array to unit vector.
+     */
+    public static void toUnit(double[] vector) {
+        toMagnitude(vector, 1.0);
+    }
+
+    /**
+     * Scale n-dimensional float array to unit vector.
+     */
+    public static void toUnit(float[] vector) {
+        toMagnitude(vector, 1.0);
+    }
+
+    /**
+     * Scale n-dimensional double array to vector with the specified magnitude.
+     */
+    public static void toMagnitude(double[] vector, double magnitude) {
+        double norm = 0.0;
+        for (int i = 0; i < vector.length; i++) {
+            norm += Math.pow(vector[i], 2);
+        }
+
+        norm = Math.sqrt(norm);
+
+        for (int i = 0; i < vector.length; i++) {
+            vector[i] = (vector[i] / (norm / magnitude));
```

Comment:
I'm not a fan of this grouping (since it indicates nested fractions).
What about: `vector[i] * magnitude / norm`?